### PR TITLE
Add Sentry sender thread

### PR DESCRIPTION
**Pull request by feymartynov**

### CODE REVIEW

File: src/extension/mod.rs [hunk 1]
```diff
@@ -13,5 +13,30 @@ mod svc_authn;
...
 mod svc_authz;
 
+/// Integration with Sentry exception tracker.
+///
+/// Setup:
+/// 1. Enable `sentry` feature in `Cargo.toml`:
+///    ```
+///    svc_error = { version = "0.1", features = ["sentry"] }
+///    ```
+/// 2. Add `sentry` section to your app's config of type `svc_error::sentry::Config`:
```
Comment (manifest) [inline src/extension/mod.rs]: misspelling

File: src/extension/sentry.rs [hunk 2]
```diff
@@ -1,10 +1,55 @@
...
+
+            if let Some(server_name) = config.server_name.clone() {
+                event.server_name = Some(Cow::from(server_name));
+            }
+
+            sentry::capture_event(event);
+        }
+    });
+
+    tx
```
Comment (manifest) [inline src/extension/sentry.rs]: There's no need to expose a `tx`

File: src/extension/mod.rs [hunk 3]
```diff
@@ -13,5 +13,30 @@ mod svc_authn;
...
+/// 3. When initializing your app call:
+///    ```
+///    let sentry_tx = svc_error::extension::sentry::init(&config.sentry);
+///    ```
+///    It spawns a thread for asynchonous error sending and returns a sender.
+/// 4. Send an `svc_error::Error` or any other type that implements `svc_error::ProblemDetails`
+///    to the `sentry_tx` whenever you like it to be sent to Sentry:
+///    ```
+///    let err = svc_error::Error::builder().detail("Something bad").build();
+///    sentry_tx.send(Box::new(err))?;
```
Comment (manifest) [inline src/extension/mod.rs]: We can create a boxed trait object internally.

File: src/lib.rs [hunk 4]
```diff
@@ -46,11 +44,43 @@ impl ProblemDetails for Error {
...
         self.set_status_code(value)
     }
+
+    fn set_detail(&mut self, detail: &str) -> &mut Self {
+        self.set_detail(detail)
+    }
 }
 
+/// Boxed ProblemDetails for sending between threads.
+pub type BoxedProblemDetails = Box<dyn ProblemDetailsReadOnly + Send>;
```
Comment (manifest) [inline src/lib.rs]: There's no need to make this type it public. We're going to use it internally only.

**Comment by manifest** [comment]
The related [PR](https://github.com/netology-group/conference/pull/49/files) for the Conference app

**Comment by manifest** [comment]
The related [PR](https://github.com/netology-group/janus-conference/pull/52) for the **Janus Conference**

File: Cargo.toml [hunk 5]
```diff
@@ -20,6 +20,9 @@ svc-agent = { version = ">=0.5", optional = true }
 svc-authn = { version = ">=0.4", optional = true }
 svc-authz = { version = ">=0.4", optional = true }
 sentry = { version = ">=0.16", optional = true }
+atom = { version = "0.3", optional = true }
+lazy_static = { version = "1.4", optional = true }
 
 [features]
 default = []
+sentry-extension = ["sentry", "atom", "lazy_static"]
```
Comment (feymartynov) [inline Cargo.toml]: Not just `sentry` because of collision with `sentry` dependency.

**Pull request merged**